### ISSUE TITLE
Width check in canvas-latex

### ISSUE THREAD

**Issue by developer-sunnywu** [CurriculumAssociates/canvas-latex#11]
Hello, 

As I want to do word wrap in canvas-latex,  Is that something that can do width checking in the canvas-latex library? Thanks.

**Comment by ry-randall** [comment]
Hi @developer-sunnywu. 

If you're trying to make the math wrap when it hits a certain width, then I'm afraid it's not currently supported. I can think about adding it as a feature, but I worry about the potential edge cases with doing something like this. What if you have a long fraction, and it exceeds the width, would part of the fraction extend to another line? Additionally this wouldn't work well with other mathematical expressions, like `\boxed`, `\sqrt`, etc.

I might be able to help better if I can understand the problem a bit more. What exactly are you trying to do? Are you using a library like CreateJS, or PixiJS?

**Comment by developer-sunnywu** [comment]
@ry-randall  Thank you for your reply.
We are currently using CreateJS. We are trying to display a maths problem. But at this stage, we are only able to calculate the width of the displayed text after rendering latex in canvas. That means if we want to do word wrap, this would be consuming a lot of CPU power. I hope the library can provide a feature like measureText() Method in canvas.
 So that when we type a new word, we can check whether it should jump to a new line. 
It would be great if it provides inline and block function just like latex. Thank you

**Comment by ry-randall** [comment]
@developer-sunnywu Sorry for the late response. I've been away for the holidays.  So, I think I understand the problem better, and fortunately I think we already have a solution for your problem. 

The library is renderer agnostic, and calculates all the necessary rendering data in a tree-like structure before attempting to render to CreateJS or PIXI. You should be able to first run the `VirtualNodeBuilder` before rendering to see the width/height/etc.

Here's an example: https://github.com/CurriculumAssociates/canvas-latex/blob/master/test/unit/tests/NodeBuilder.test.js#L24-L28

The result is a big JS object, but I think the result you care about is: https://github.com/CurriculumAssociates/canvas-latex/blob/master/test/unit/tests/__snapshots__/NodeBuilder.test.js.snap#L5605-L5609 

This allows access to the tree's root bounds, including width/height/etc. 

So in other words, you can use the `VirtualNodeBuilder` to check the width, and render it if it fits.

Let me know if you run into any issues.

**Comment by ry-randall** [comment]
@rfandl-cainc Would you mind closing this? Looks like I don't have write privs anymore 😕

**Comment by developer-sunnywu** [comment]
Ok. I just saw this. Thank you very much
